src/dotnet/Telemetry.cs

```diff
@@ -9,18 +10,28 @@ namespace Microsoft.DotNet.Cli
 {
     public class Telemetry : ITelemetry
     {
+        public const double TimeoutInMilliseconds = 100;
+
         private bool _isInitialized = false;
         private TelemetryClient _client = null;
 
         private Dictionary<string, string> _commonProperties = null;
         private Dictionary<string, double> _commonMeasurements = null;
+        private Task _trackEventTask = null;
+
+        private const int ReciprocalSampleRateValue = 5;
```

Comment:
You make a great point regarding the sampling potential gaps.  This was originally requested because of original performance concerns, but after having a recent meeting, the issue might be resolved and sampling might not be needed.